ee/tabby-webserver/src/handler.rs

```diff
@@ -28,10 +29,11 @@ pub async fn attach_webserver(
     local_port: u16,
 ) -> (Router, Router) {
     let ctx = create_service_locator(logger, code, is_chat_enabled).await;
-    cron::run_cron(ctx.auth(), ctx.job(), ctx.worker(), local_port).await;
-
     let repository_cache = Arc::new(RepositoryCache::new_initialized(ctx.repository()).await);
-    repository_cache.start_reload_job().await;
+    let events = cron::run_cron(ctx.auth(), ctx.job(), ctx.worker(), local_port).await;
+    if let Some(events) = events {
+        repository_cache.start_reload_listener(&events);
+    }
```

Comment:
```suggestion
    let events = cron::run_cron(ctx.auth(), ctx.job(), ctx.worker(), local_port).await;
    let repository_cache = RepositoryCache::new_initialized(ctx.repository(), events. scheduler_job_succeeded).await
```

Reply:
Unless we want to be using `.unwrap()` on `run_cron`, we can't do this

Reply:
`.await?`